blackjax/rwmh.py

```diff
@@ -0,0 +1,124 @@
+"""Public API for the Random Walk Metropolis-Hastings Kernel"""
+from typing import Callable, NamedTuple
+
+import jax.numpy as jnp
+import jax.random
+from jax.flatten_util import ravel_pytree
+
+import blackjax.inference.metrics as metrics
+from blackjax.common import Array, PyTree
+
+__all__ = ["new_state", "kernel", "RWMHState"]
+
+
+class RWMHState(NamedTuple):
+    """ State of the RWMH chain
+
+    position
+        Current position of the chain.
+    potential_energy
+        Current value of the potential energy
+    """
+    position: PyTree
+    potential_energy: float
+
+
+class RWMHInfo(NamedTuple):
+    """Additional information on the RWMH chain.
+
+    This additional information can be used for debugging or computing
+    diagnostics.
+
+    acceptance_probability
+        The acceptance probability of the transition, linked to the energy
+        difference between the original and the proposed states.
+    is_accepted
+        Whether the proposed position was accepted or the original position
+        was returned.
+    proposal
+        The state proposed by the proposal.
+    """
+
+    acceptance_probability: float
+    is_accepted: bool
+    proposal: PyTree
+
+
+def new_state(position: PyTree, potential_fn: Callable) -> RWMHState:
+    """Create a chain state from a position.
+
+    Parameters:
+    -----------
+    position: PyTree
+        The initial position of the chain
+    potential_fn: Callable
+        Target potential function of the chain
+    """
+    return RWMHState(position, potential_fn(position))
+
+
+def kernel(
+        potential_fn: Callable,
+        inverse_mass_matrix: Array,
```

Comment:
We want to pass a `proposal_generator` instead of an `inverse_mass_matrix` here. We don't want to limit the algorithm to moves drawn from a normal distribution.

Reply:
I now mostly addressed this, which made a lot of the above logic disappear. What do you think about it?